### PR TITLE
mv findVtblIndex() and overrideInterface() to funcsem.d

### PR DESCRIPTION
**Pull request by WalterBright**
also changed the interface to findVtblIndex()

otherwise cut & paste

### CODE REVIEW

File: compiler/src/dmd/declaration.h [hunk 1]
```diff
@@ -703,7 +703,7 @@ class FuncDeclaration : public Declaration
     Expressions *fdensureParams(Expressions *fdep);
     bool equals(const RootObject * const o) const override final;
 
-    int findVtblIndex(Dsymbols *vtbl, int dim);
+    int findVtblIndex(FuncDeclaration *fd, DArray<Dsymbol*> vtbl);
```
Comment (ibuclaw) [inline compiler/src/dmd/declaration.h:706]: This looks wrong to me. It's a free function now.

File: compiler/src/dmd/funcsem.d [hunk 2]
```diff
@@ -1217,3 +1217,152 @@ extern (D) bool checkForwardRef(FuncDeclaration fd, const ref Loc loc)
...
+ * Find index of function in vtbl[0..length] that
+ * this function overrides.
+ * Prefer an exact match to a covariant one.
+ * Params:
+ *      vtbl     = vtable to use
+ * Returns:
+ *      -1      didn't find one
+ *      -2      can't determine because of forward references
+ */
+int findVtblIndex(FuncDeclaration fd, Dsymbol[] vtbl)
```
Comment (ibuclaw) [inline compiler/src/dmd/funcsem.d:1231]: Missing extern(C++)

**Review by ibuclaw — DISMISSED** [review]

File: compiler/src/dmd/funcsem.d (hunk 2 above)
Comment (ibuclaw) [reply compiler/src/dmd/funcsem.d:1231]: Say that, wait. I should check whether this is actually used by the C++ backends first.

Comment (ibuclaw) [reply compiler/src/dmd/funcsem.d:1231]: It's unused,leave as extern(D)

File: compiler/src/dmd/declaration.h (hunk 1 above)
Comment (WalterBright) [reply compiler/src/dmd/declaration.h]: deleted

**Review by ibuclaw — APPROVED** [review]

**Review by RazvanN7 — APPROVED** [review]

File: compiler/src/dmd/funcsem.d (hunk 2 above)
Comment (RazvanN7) [inline compiler/src/dmd/funcsem.d:1231]: Need to also update the documentation, hence the doc tester failure:

`1msrc/dmd/funcsem.d(1231): [1;33mWarning: [mDdoc: parameter count mismatch, expected 2, got 1`

Comment (WalterBright) [reply compiler/src/dmd/funcsem.d:1232]: updated

**Pull request merged**